modules/codegen-plugin/src/smithy4s/codegen/Smithy4sCodegenPlugin.scala

```diff
@@ -252,36 +260,51 @@ object Smithy4sCodegenPlugin extends AutoPlugin {
       }
     },
     config / smithy4sRenderOptics := false,
+    config / smithy4sRenderValidatedNewtypes := false,
     config / smithy4sGeneratedSmithyMetadataFile := {
       (config / sourceManaged).value / "smithy" / "generated-metadata.smithy"
     },
     config / smithy4sGeneratedSmithyFiles := {
       val cacheFactory = (config / streams).value.cacheStoreFactory
-      val cached = Tracked.inputChanged[(String, Boolean), Seq[File]](
-        cacheFactory.make("smithy4sGeneratedSmithyFilesInput")
-      ) { case (changed, (wildcardArg, shouldGenerateOptics)) =>
-        val lastOutput = Tracked.lastOutput[Boolean, Seq[File]](
-          cacheFactory.make("smithy4sGeneratedSmithyFilesOutput")
-        ) { case (changed, prevResult) =>
-          if (changed || prevResult.isEmpty) {
-            val file = (config / smithy4sGeneratedSmithyMetadataFile).value
-            IO.write(
-              file,
-              s"""$$version: "2"
-                 |metadata smithy4sWildcardArgument = "$wildcardArg"
-                 |metadata smithy4sRenderOptics = $shouldGenerateOptics
-                 |""".stripMargin
-            )
-            Seq(file)
-          } else {
-            prevResult.get
+      val cached =
+        Tracked
+          .inputChanged[(String, Boolean, Boolean), Seq[File]](
+            cacheFactory.make("smithy4sGeneratedSmithyFilesInput")
+          ) {
+            case (
+                  changed,
+                  (
+                    wildcardArg,
+                    shouldGenerateOptics,
+                    shouldRenderValidatedNewtypes
+                  )
+                ) =>
+              val lastOutput = Tracked.lastOutput[Boolean, Seq[File]](
+                cacheFactory.make("smithy4sGeneratedSmithyFilesOutput")
+              ) { case (changed, prevResult) =>
+                if (changed || prevResult.isEmpty) {
+                  val file =
+                    (config / smithy4sGeneratedSmithyMetadataFile).value
+                  IO.write(
+                    file,
+                    s"""$$version: "2"
+                       |metadata smithy4sWildcardArgument = "$wildcardArg"
+                       |metadata smithy4sRenderOptics = $shouldGenerateOptics
+                       |metadata smithy4sRenderValidatedNewtypes = $shouldRenderValidatedNewtypes
+                       |""".stripMargin
+                  )
+                  Seq(file)
+                } else {
+                  prevResult.get
+                }
+              }
+              lastOutput(changed)
```

Comment:
**suggestion:** I'd extract some of this to avoid the current nesting level, which is becoming a bit unwieldy

Reply:
I don't want build-plugin settings to control rendering behaviour.